tools/azure-sdk-tools/devtools_testutils/azure_testcase.py

```diff
@@ -100,6 +100,23 @@ def _get_replay_processors(self):
     def is_playback(self):
         return not self.is_live
 
+    def get_settings_value(self, key):
+        key_value = os.environ.get("AZURE_"+key, None)
+
+        if key_value and self._real_settings and getattr(self._real_settings, key) != key_value:
+            raise ValueError("You have both AZURE_{key} env variable and mgmt_settings_real.py for {key} to difference values".format(key=key))
+
+        if not key_value:
+            key_value = getattr(self.settings, key)
+        return key_value
+
+    def set_value_to_scrub(self, key, default_value):
+        if self.is_live:
+            value = self.get_settings_value(key)
+            self.scrubber.register_name_pair(value, default_value)
```

Comment:
Looks like this should return `value or default_value`.

Reply:
Let's return `value`, in live mode you should never need the default_value in your testcase.